ssl/ssl_test.cc

```diff
@@ -686,6 +727,38 @@ static const HybridGroupTest kHybridGroupTests[] = {
       KYBER768_R3_CIPHERTEXT_BYTES,  // accept_share_sizes[1]
     },
   },
+  {
+    NID_SecP256r1MLKEM768,
+    SSL_GROUP_SECP256R1_MLKEM768,
+    P256_KEYSHARE_SIZE + MLKEM768_PUBLIC_KEY_BYTES,
+    P256_KEYSHARE_SIZE + MLKEM768_CIPHERTEXT_BYTES,
+    P256_SECRET_SIZE + MLKEM768_SHARED_SECRET_LEN,
+    {
+      P256_KEYSHARE_SIZE,             // offer_share_sizes[0]
+      MLKEM768_PUBLIC_KEY_BYTES,      // offer_share_sizes[1]
+    },
+    {
+      P256_KEYSHARE_SIZE,             // accept_share_sizes[0]
+      MLKEM768_CIPHERTEXT_BYTES,      // accept_share_sizes[1]
+    },
+  },
+  {
+    NID_X25519MLKEM768,
+    SSL_GROUP_X25519_MLKEM768,
+    X25519_KEYSHARE_SIZE + MLKEM768_PUBLIC_KEY_BYTES,
+    X25519_KEYSHARE_SIZE + MLKEM768_CIPHERTEXT_BYTES,
+    X25519_SECRET_SIZE + MLKEM768_SHARED_SECRET_LEN,
+    {
+      // MLKEM768 is sent first for X25519MLKEM768 for FIPS compliance
```

Comment:
Why is the order of the key shares important for FIPS?

Reply:
```
Both groups enable the derivation of TLS session keys using FIPS-approved schemes.
NIST's special publication 800-56Cr2 SP56C approves the usage of HKDF with two
distinct shared secrets, with the condition that the first one is computed by a FIPS-
approved key-establishment scheme. FIPS also requires a certified implementation of 
the scheme, which will remain more ubiqutous for secp256r1 in the coming years. For 
this reason we put the ML-KEM-768 shared secret first in X25519MLKEM768, and the 
secp256r1 shared secret first in SecP256r1MLKEM768.
```
https://datatracker.ietf.org/doc/html/draft-kwiatkowski-tls-ecdhe-mlkem-02#name-negotiated-groups